provider/machine_pool_resource.go

```diff
@@ -455,30 +455,29 @@ func (r *MachinePoolResource) Update(ctx context.Context, request tfsdk.UpdateRe
 	response.Diagnostics.Append(diags...)
 }
 
-func getSpotInstances(state *MachinePoolState, mpBuilder *cmv1.MachinePoolBuilder) (
-	useSpotInstances bool, errMsg string) {
-	useSpotInstances = false
+func setSpotInstances(state *MachinePoolState, mpBuilder *cmv1.MachinePoolBuilder) error {
+	useSpotInstances := !state.UseSpotInstances.Unknown && !state.UseSpotInstances.Null && state.UseSpotInstances.Value
```

Comment:
```suggestion
	useSpotInstances := !helper.IsStringAttributeEmpty(state.UseSpotInstances)
```

Reply:
but useSpotInstances is boolean, not string